git-sync/main.go

```diff
@@ -213,3 +221,27 @@ func setupGitAuth(username, password, gitURL string) error {
 
 	return nil
 }
+
+func setupGitSSH() error {
+	log.Println("setting up git SSH credentials")
+
+	if _, err := os.Stat("/etc/git-secret/id-rsa"); err != nil {
+		log.Fatal("error: could not find SSH secret at /etc/git-secret/id-rsa")
+	}
+
+	if err := os.Mkdir("/root/.ssh", 0600); err != nil {
```

Comment:
Does $HOME work more generally?  Especially if you mkdirall 755.  All of these problems would go away if you used a wrapper ssh script.  You could even manipulate the ssh exe and script on disk in your container build.